src/drivers/distance_sensor/tfmini/tfmini_parser.cpp

```diff
@@ -58,44 +57,79 @@ const char *parser_state[] = {
 	"5_GOT_STRENGTH_L",
 	"6_GOT_STRENGTH_H",
 	"7_GOT_PRESERVED",
-	"8_GOT_QUALITY"
-	"9_GOT_CHECKSUM"
+	"8_GOT_QUALITY",
+	"9_GOT_CHECKSUM",
+	"10_GOT_COMMANDRESPONSE"
 };
 #endif
 
-int tfmini_parse(char c, char *parserbuf, unsigned *parserbuf_index, TFMINI_PARSE_STATE *state, float *dist)
+int tfmini_parse(uint8_t c, uint8_t *parserbuf, unsigned *parserbuf_index, TFMINI_PARSE_STATE *state,
+		 TFMINI_MODEL hw_model, float *dist, uint16_t *strength, float *temperature, uint8_t *commandresponse,
+		 uint8_t *commandresponse_size)
 {
 	int ret = -1;
 	//char *end;
+	unsigned char cksm = 0;
 
 	switch (*state) {
 	case TFMINI_PARSE_STATE::STATE6_GOT_CHECKSUM:
-		if (c == 'Y') {
+	case TFMINI_PARSE_STATE::STATE8_GOT_RESPONSECHECKSUM:
+		if (c == 0x59) {
```

Comment:
Could you abstract all of these hex values as #defines or static constexpr vars with names that indicate what they do?